lib/argo_cd/client.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+require_relative 'response'
+
+module ArgoCd
+  # Client class for interacting with ArgoAPI
+  class Client
+    attr_accessor :app_id, :deployment_name, :current_user_id
+
+    def initialize(app_id, deployment_name, current_user_id)
+      @app_id = app_id
+      @deployment_name = deployment_name
+      @current_user_id = current_user_id
+    end
+
+    def app_info
+      uri = URI("#{base_path}/api/v1/applications?name=#{deployment_name}")
+
+      if connected_app.blank? || token_expired?(connected_app)
+        token_response = generate_token
+        return token_response unless token_response.successful?
+      end
+
+      get_app_info(uri)
+    end
+
+    def get_app_info(uri)
+      https = Net::HTTP.new(uri.host, uri.port)
+      https.verify_mode = OpenSSL::SSL::VERIFY_NONE if Rails.env.development?
+      response = https.get(uri.path, request_headers)
+
+      Response.new(response: response)
+    end
+
+    def request_headers
+      {
+        'Authorization' => "Bearer #{jwt_token}",
+        'Content-Type' => 'application/json'
+      }
+    end
+
+    def current_deploy
+      # TODO: - dig in after first iteration
+    end
+
+    private
+
+    def base_path
+      return 'https://localhost:8080' unless Rails.env.production?
+
+      ENV['ARGO_API_BASE_PATH']
+    end
+
+    def connected_app
+      @connected_app ||= ConnectedApp.find_by(user_id: current_user_id, app_id: app_id)
+    end
+
+    def jwt_token
```

Comment:
Could you change this to just `jwt`? This reads as `json web token token` 😄 .
```suggestion
    def jwt
```